### ISSUE TITLE
Using localhost and trying to switch rooms

### ISSUE THREAD

**Issue by dejanzivanov** [thecodingmachine/workadventure#907]
Hello, I am having a problem with changing the rooms using my localhost, the issue is when I am using my local machine for testing purpose there is a problem with a directory and the designated map is not found.

**Comment by moufmouf** [comment]
The problem seems to occur because when leaving a map for anothe map, the port is lost.

So:
https://play.workadventu.re/_/global/localhost:8080/map.json

becomes
https://play.workadventu.re/_/global/localhost/other_map.json

We need to fix the missing ":8080".

**Comment by jonnytest1** [comment]
i also seem to get some protocol issues when switching between maps

**Comment by dejanzivanov** [comment]
Hello @jonnytest1 , at the moment if you are testing your local environment. I would give you suggestion when you are switching between the maps, you absolute link of your github file location, for example: 

https://play.workadventu.re/_/global/USERNAME.github.io/NAME_OF_YOUR_REPO/NAME_OF_MAP.json

This worked out perfectly for me as work around. By the way Work Adventure should be as published page on your repository.

Or if you are using your local environment, you can just add ":8080" after "localhost" before the slash "/other_map.json". 


Hopefully this helps you!
Best regards :)

**Comment by jonnytest1** [comment]
well im not gonna wait 10 minute for the github cache to resolve every time i do some minor changes :D - and my local files are server from my local nginx hich runs on port 80 anyways :) so in that regard there are no problems

**Comment by Flote37** [comment]
I had the same problem on my local environment.
To reproduce : 
- Load a map
- Exit the map through one exit
- Come back to the first map through the appropriate exit => black screen

TL;DR : To fix this, I enabled HTTPS (with the https-localhost module) on my localhost and now it works fine.

If it can help someone who can dig deeper, this is what I understood about the problem :

The first time a map is loaded it generates two requests for the map : 1 with HTTPS (which fails since I'm not running on HTTPS) and then, as a fallback, 1 with HTTP (which succeed).
The second time (when you come back to an already visited map), I find only the HTTPS request which fails and display a black screen.

I found some code that replace HTTPS with HTTP but it is executed in both case :
GameScene.ts
```javascript
this.load.on(FILE_LOAD_ERROR, (file: { src: string }) => {
            // If we happen to be in HTTP and we are trying to load a URL in HTTPS only... (this happens only in dev environments)
...
if (
    window.location.protocol === "https:" &&
    file.src === this.MapUrlFile &&
    (host === "127.0.0.1" || host === "localhost" || host.endsWith(".localhost")) &&
    this.originalMapUrl === undefined
)
...
this.MapUrlFile = this.MapUrlFile.replace("https://", "http://");
``` 
I don't know how to debug further than that for now :(
